hydroflow_lang/src/graph/ops/cross_join.rs

```diff
@@ -16,6 +15,25 @@ use syn::parse_quote;
 /// source_iter(vec!["dog", "cat"]) -> [1]my_join;
 /// my_join -> for_each(|(v1, v2)| println!("({}, {})", v1, v2));
 /// ```
+///
+/// `cross_join` can also be provided with one or two generic lifetime persistence arguments
+/// in the same was as [`join`](#join), see [`join`'s documentation](#join) for more info.
+///
+/// ```rustbook
+/// let (input_send, input_recv) = hydroflow::util::unbounded_channel::<&str>();
+/// let mut flow = hydroflow::hydroflow_syntax! {
+///     my_join = cross_join::<'tick>();
+///     source_iter(["hello", "bye"]) -> [0]my_join;
+///     source_stream(input_recv) -> [1]my_join;
+///     my_join -> for_each(|(s, t)| println!("({}, {})", s, t));
+/// };
+/// input_send.send("oakland").unwrap();
+/// flow.run_tick();
+/// input_send.send("san francisco").unwrap();
+/// flow.run_tick();
+/// ```
+/// Prints only `"(hello, oakland)"` and `"(bye, oakland)"`. The `source_iter` is only included in
+/// the first tick, then forgotten.
```

Comment:
Not sure if we ought to duplicate the same docs as `join` but at least wanted the doctest in here, primarily for testing rather than documentation. could move to a non-doc test

Reply:
seems fine to ref the join docs